Shallow_water_approximate.ipynb

```diff
@@ -15,8 +15,24 @@
     "\\begin{align}\n",
     "    h_t + (hu)_x & = 0 \\label{SWA_mass} \\\\\n",
     "    (hu)_t + \\left(hu^2 + \\frac{1}{2}gh^2\\right)_x & = 0. \\label{SWA_mom}\n",
-    "\\end{align}  \n",
-    "We investigated the exact solution of the Riemann problem for this system in [Shallow_water.ipynb](Shallow_water.ipynb); code for the exact solution is in [exact_solvers/shallow_water.py](exact_solvers/shallow_water.py).  We found that the system has two genuinely nonlinear characteristic fields, so the solution consists of two waves, each of which may be a shock or a rarefaction.\n",
+    "\\end{align} "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To examine the Python code for this chapter, and for the exact Riemann solution, see:\n",
+    "\n",
+    " - [exact_solvers/shallow_water.py](exact_solvers/shallow_water.py) ...\n",
+    "   [on github.](https://github.com/clawpack/riemann_book/blob/master/exact_solvers/shallow_water.py)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We investigated the exact solution of the Riemann problem for this system in [Shallow_water.ipynb](Shallow_water.ipynb).  found that the system has two genuinely nonlinear characteristic fields, so the solution consists of two waves, each of which may be a shock or a rarefaction.\n",
```

Comment:
I think the word "We" was accidentally deleted here.  Maybe just end the previous sentence with a comma and insert "and"?